### ISSUE TITLE
[Inconsistent]: Agent installed with System integration gets offline-healthy when assigned to policy with Endpoint Security. 

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#30329]
**Kibana version:**  8.0.0 BC-1 Kibana cloud environment

**Host OS and Browser version:** Windows, All

**Build details:**
```
BUILD: 49192
COMMIT: 57ca5e139a33dd2eed927ce98d8231a1f217cd15
Artifact Link: https://staging.elastic.co/8.0.0-da04e908/summary-8.0.0.html
```

**Preconditions:**
1. 8.0.0 BC-1 Kibana cloud environment should be available.
2. Test Signing should be off.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Install 8.0 BC-1 agent with Default policy having System integration.
3. Create a new policy with System and Endpoint Security integrations, say `NEW POLICY`.
4. Assign agent to this `NEW POLICY` policy.
5. Observe agent goes offline and online.
6. Observe agent is not available under Endpoints tab.
7. Observe `Endpoint` folder is not available under installation path `C:\Program Files\Elastic\`.

**Logs:**
[logs.zip](https://github.com/elastic/endpoint-dev/files/8021460/logs.zip)

**Expected Result:**
Agent installed with System integration should remain Healthy when assigned to policy with Endpoint Security.

**Screenshots:**
![4](https://user-images.githubusercontent.com/77374876/152934460-4cbe0b4f-08bd-4097-9be8-c9c6b4ad0370.JPG)
![5](https://user-images.githubusercontent.com/77374876/152934472-071dfdad-accb-4891-b6ea-d38f16b82a5f.JPG)

**NOTE:**
-  Validated on Linux ARM artifacts and is reproducible there too.

**Comment by amolnater-qasource** [comment]
Further, It's not an upgraded environment we have created 8.0 BC-1 from staging platform.
Please find below policy to which agent was assigned.
[elastic-agent.zip](https://github.com/elastic/endpoint-dev/files/8030813/elastic-agent.zip)

We still have an agent installed with name `mainqa-atlcolo-10-0-7-238` on this environment with this issue reproduced.
**Error logs:**
```
13:50:46.316
elastic_agent
[elastic_agent][error] Could not communicate with fleet-server Checking API will retry, error: status code: 400, fleet-server returned an error: BadRequest
```

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Reviewed & mention to @ferullo

CC: @jlind23

**Comment by jlind23** [comment]
@amolnater-qasource http 400 seems to be related to a falsy token. Is it possible that the installed agent has faulty credentials? Could it be possible to reinstall it?

**Comment by amolnater-qasource** [comment]
Hi @jlind23 
>http 400 seems to be related to a falsy token. Is it possible that the installed agent has faulty credentials? Could it be possible to reinstall it?

As the agent was installed healthy, we don't think if agent has faulty credentials as agent installation only requires enrollment token and fleet server url while installation. If any of this value is not correct, agent doesn't get installed.

We have attempted various times on that particular environment shared on endpoint-dev ticket.

We found that inconsistent issue is reproducible on following below step:
- Installing agent with default policy with system integration and then assigning agent to a policy with Endpoint Security.

At the time of reporting the actual issue, the issue was reproducible on 02 different environments with various agents.

However later it wasn't reproducible on the new environments[with the same commits].

Thanks

**Comment by michel-laterman** [comment]
I was unable to recreate the issue on 8.0 using different linux VMs as the base

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
This issue was inconsistent. We have shared our environment at https://github.com/elastic/endpoint-dev/issues/10494#issuecomment-1033435394

Further please find below logs for agent:
[logs.zip](https://github.com/elastic/endpoint-dev/files/8021460/logs.zip)

Please let us know if anything else is required from our end.
Thanks

**Comment by michel-laterman** [comment]
Are there any logs from the fleet-server?
Can you recreate this with logs set to debug?

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
There are no fleet-server logs.
Please find new debug level logs for this issue:
[logs.zip](https://github.com/elastic/beats/files/8078057/logs.zip)

We have reproduced this issue again on the above shared environment.
Please let us know if that environment is still required or we can delete it.

Thanks

**Comment by michel-laterman** [comment]
Are you able to recreate this with the released version of 8.0?
I can't reproduce it on that version and need to see the error message from fleet server in order to figure out what's going wrong with the requests

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
We have revalidated this on 8.0 released and 8.0.1 BC-1 and **didn't find it reproducible** now.
- Agent installed with System integration should remain Healthy when assigned to policy with Endpoint Security.

**8.0.1 BC-1 Build details:**
```
BUILD: 49342
COMMIT: f4b44d7eb7355c9d1e38d9f2dc753b3fe10c601c
Artifact Link: https://staging.elastic.co/8.0.1-91daef6b/downloads/beats/elastic-agent/elastic-agent-8.0.1-windows-x86_64.zip
```

Hence we are **closing** this.
Further we will keep track of this inconsistent issue in our testing on later builds.
Thanks

**Comment by RichiCoder1** [comment]
Ran into this when using the Okta Logs integration with 7.14. Agent will enroll and then immediately start failing w/ 400 requests back to the Fleet Server with no logs indicating anything went wrong.
